### PR TITLE
Python 3 Upgrade

### PR DESCRIPTION
**Pull request by msaysell**
- Lots of removal of future imports
- Upgraded dependencies
- Upgraded circleci to use Couchbase 2.5.0
- Fixing of tests
- Removed test that was just testing test setup (inception?)
- Unicode-ing

### CODE REVIEW

File: tests/stats/test_retrieve.py [hunk 1]
```diff
@@ -12,28 +12,26 @@ def expected_stats(syncgateway_version_str):
...
-        'syncGateway_index_clocks',
-        'syncGateway_rest',
-        'syncGateway_stats',
+        "cb",
+        "cmdline",
+        "mc",
+        "memstats",
+        "syncgateway",
+        "syncGateway_changeCache",
+        "syncGateway_dcp",
```
Comment (ChrisDouglasBiosite) [inline tests/stats/test_retrieve.py]: Why has this list changed?
Was it wrong previously?

File: tests/test_fixtures.py [hunk 2]
```diff
@@ -1,30 +0,0 @@
-from __future__ import absolute_import, print_function, unicode_literals
```
Comment (ChrisDouglasBiosite) [inline tests/test_fixtures.py]: What is the reason behind deleting these tests?

File: tests/stats/test_retrieve.py (hunk 1 above)
Comment (ChrisDouglasBiosite) [reply tests/stats/test_retrieve.py]: This seems to be the reason why the 1.5 build is failing.

File: tests/test_fixtures.py (hunk 2 above)
Comment (msaysell) [reply tests/test_fixtures.py]: When python 3 was first released, python introduced future functions into python 2.x to help projects transition to the new version but still support python 2, we don't need the support anymore

**Comment by jamescooke** [comment]
@msaysell Updating for Python 3 is great :+1: 

I can see that you have multiple commits fixing lint. Both `isort` and `yapf` can write back to project files to fix problems they find. The `fixlint` recipe (https://github.com/constructpm/pysyncgateway/blob/master/Makefile#L62-L67) should fix everything for you in one go which you can then commit.

Also by removing the `from __future__` imports, Python 2 will no longer be supported. Therefore I would recommend adding `python_requires='>=3.5, <4',` or similar to `setup.py`. This will help `pip` find suitable versions for people still running Python 2. Also the `"Programming Language :: Python :: 2"` and `2.7` classifiers should probably also be removed.

Hope that's helpful.

**Comment by msaysell** [comment]
@jamescooke - Thank you for your suggestions!

`fixlint` did indeed help alot with the seemingly neverending lint issues.

Also updated the setup.py to remove the last of the python 2.7 support!

**Pull request merged**